packages/inference/src/providers/fal-ai.ts

```diff
@@ -14,109 +14,232 @@
  *
  * Thanks!
  */
+import type { AutomaticSpeechRecognitionOutput } from "@huggingface/tasks";
 import { InferenceOutputError } from "../lib/InferenceOutputError";
 import { isUrl } from "../lib/isUrl";
-import type { BodyParams, HeaderParams, InferenceTask, ProviderConfig, UrlParams } from "../types";
+import type { BodyParams, HeaderParams, InferenceTask, UrlParams } from "../types";
 import { delay } from "../utils/delay";
+import { omit } from "../utils/omit";
+import { TaskProviderHelper } from "./providerHelper";
 
-const FAL_AI_API_BASE_URL = "https://fal.run";
-const FAL_AI_API_BASE_URL_QUEUE = "https://queue.fal.run";
+export interface FalAiQueueOutput {
+	request_id: string;
+	status: string;
+	response_url: string;
+}
 
-const makeBaseUrl = (task?: InferenceTask): string => {
-	return task === "text-to-video" ? FAL_AI_API_BASE_URL_QUEUE : FAL_AI_API_BASE_URL;
-};
+interface FalAITextToImageOutput {
+	images: Array<{
+		url: string;
+	}>;
+}
 
-const makeBody = (params: BodyParams): Record<string, unknown> => {
-	return params.args;
-};
+interface FalAIAutomaticSpeechRecognitionOutput {
+	text: string;
+}
 
-const makeHeaders = (params: HeaderParams): Record<string, string> => {
-	return {
-		Authorization: params.authMethod === "provider-key" ? `Key ${params.accessToken}` : `Bearer ${params.accessToken}`,
+interface FalAITextToSpeechOutput {
+	audio: {
+		url: string;
+		content_type: string;
 	};
-};
+}
+export const FAL_AI_SUPPORTED_BLOB_TYPES = ["audio/mpeg", "audio/mp4", "audio/wav", "audio/x-wav"];
 
-const makeUrl = (params: UrlParams): string => {
-	const baseUrl = `${params.baseUrl}/${params.model}`;
-	if (params.authMethod !== "provider-key" && params.task === "text-to-video") {
-		return `${baseUrl}?_subdomain=queue`;
+abstract class FalAITask extends TaskProviderHelper {
+	constructor(task: InferenceTask, url?: string) {
+		super("fal-ai", url || "https://fal.run", task);
 	}
-	return baseUrl;
-};
 
-export const FAL_AI_CONFIG: ProviderConfig = {
-	makeBaseUrl,
-	makeBody,
-	makeHeaders,
-	makeUrl,
-};
+	preparePayload(params: BodyParams): Record<string, unknown> | Promise<Record<string, unknown>> {
```

Comment:
Why this weird return type?
```suggestion
	preparePayload(params: BodyParams): Record<string, unknown> {
```

Reply:
🤦‍♀️